geomesa-convert/geomesa-convert-avro/src/main/scala/org/locationtech/geomesa/convert/avro/AvroSimpleFeatureConverter.scala

```diff
@@ -49,4 +57,52 @@ class AvroSimpleFeatureConverter(avroSchema: Schema,
     Seq(Array(bytes, reader.read(recordReuse, decoder)))
   }
 
+  override def process(is: InputStream, ec: EvaluationContext = createEvaluationContext()): Iterator[SimpleFeature] = {
+    decoder = DecoderFactory.get.binaryDecoder(is, null)
+
+    class FeatureItr extends Iterator[SimpleFeature] {
+      private var cur: SimpleFeature = null
+
+      override def hasNext: Boolean = {
+        if (cur == null) {
+          do { fetchNext() } while (cur == null && !decoder.isEnd)
+          cur != null
+        } else {
+          true
+        }
+      }
+
+      override def next(): SimpleFeature = {
+        hasNext
+        if (cur != null) {
+          val ret = cur
+          cur = null
+          ret
+        } else throw new NoSuchElementException
+      }
+
+      def fetchNext() = {
+        if (!decoder.isEnd) {
+          ec.counter.incLineCount()
+          val rec = reader.read(null, decoder)
```

Comment:
only if its really bad and a malformed binary file...i would expect the converter to blow out and stop completely and kill the mapper/ingest thread at this point...so i guess letting it throw is ok